implementations/rust/ockam/ockam_api/src/kafka/portal_worker.rs

```diff
@@ -255,16 +257,13 @@ impl KafkaPortalWorker {
     async fn intercept_and_transform_messages(
         &mut self,
         context: &mut Context,
-        encoded_message: &Vec<u8>,
+        encoded_message: &[u8],
     ) -> Result<Option<Bytes>, InterceptError> {
         let mut encoded_buffer: Option<BytesMut> = None;
 
         for complete_kafka_message in self
             .decoder
-            .decode_messages(
-                BytesMut::from(encoded_message.as_slice()),
-                self.max_message_size,
-            )
+            .decode_messages(BytesMut::from(encoded_message), self.max_message_size)
             .map_err(InterceptError::Ockam)?
         {
             let transformed_message = match self.receiving {
```

Comment:
Is there a reason to return a value from that function? If I'm not mistaken, we always return the argument back

Reply:
Also, would it be possible to return messages from this function one by one instead of putting them in one large buffer?